### PR TITLE
[WIP] Attempt to make job submission async

### PR DESCRIPTION
**Pull request by oshadura**
Probably fixes: #470 
cc: @bbockelm

### CODE REVIEW

File: dask_jobqueue/core.py [hunk 1]
```diff
@@ -302,8 +302,8 @@ def job_file(self):
             yield fn
 
     async def _submit_job(self, script_filename):
-        # Should we make this async friendly?
-        return self._call(shlex.split(self.submit_command) + [script_filename])
+        call = shlex.split(self.submit_command + [script_filename])
+        return await asyncio.get_running_loop().run_in_executor(None, self._call(call))
```
Comment (mivade) [inline dask_jobqueue/core.py:306]: If I'm reading this right, I don't think there's a need to run this in an executor as you already made `_call` async.

**Comment by lesteve** [comment]
Some CI error looks genuine for example this one: https://github.com/dask/dask-jobqueue/pull/473/checks?check_run_id=1406501917.

I am not an async expert, but I would guess you need to add some `await` where `_call` is called, maybe changing the functions calling `_call` to `async`. One such function seems to be `_job_id_from_submit_output`, there may be others. In an ideal world, the main user-facing functions would stay synchronous so that if you get an exception it blows up in an obvious way rather than having some log message saying that an exception was never retrieved.

Full disclosure : unfortunately I don't really understand the original issue #470 ...

**Comment by guillaumeeb** [comment]
This PR is a bit old... @oshadura are you still around and up for this change?

I'm going to close this one next time I'll take a look at the activity here, unless someone updates it.

**Comment by oshadura** [comment]
@guillaumeeb sorry I was working on other things and thanks for reminder I will definitely try to take a look in next couple of days and update/close PR.

**Comment by jacobtomlinson** [comment]
It's been a long time since this PR had any activity so I'm going to close it out.

**Pull request closed without merging**